src/webparts/LaserficheRepositoryAccessWebPart/components/RepositoryViewWebPart.tsx

```diff
@@ -0,0 +1,704 @@
+          {validationError}
+          <div className='form-group row mb-3'>
+            <label className='col-sm-2 col-form-label'>Name</label>
+            <div className='col-sm-10'>
+              <input
+                type='text'
+                className='form-control'
+                id='uploadFileID'
+                onChange={SetNewFileName}
+                value={fileName}
+              />
+            </div>
+          </div>
+          <div className='card'>
+            <div
+              className={`lf-component-container${
+                adhocDialogOpened ? ' lf-adhoc-min-height' : ''
+              }`}
+            >
+              <lf-field-container
+                collapsible='true'
+                startCollapsed='true'
+                ref={fieldContainer}
+              />
+            </div>
+          </div>
+        </div>
+        <div className='modal-footer'>
+          <button
+            type='button'
+            className='btn btn-primary btn-sm'
+            onClick={ImportFileToRepository}
+          >
+            OK
+          </button>
+          <button
+            type='button'
+            className='btn btn-secondary btn-sm'
+            onClick={CloseImportFileModal}
+          >
+            Cancel
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function CreateFolderModal(props: {
+  repoClient: IRepositoryApiClientExInternal;
+  closeCreateFolderModal: () => void;
+  parentItem: LfRepoTreeNode;
+}) {
+  const [folderName, setFolderName] = React.useState('');
+  const [
+    createFolderNameValidationMessage,
+    setCreateFolderNameValidationMessage,
+  ] = React.useState<string | undefined>(undefined);
+
+  const CloseNewFolderModal = () => {
+    setCreateFolderNameValidationMessage(undefined);
+    setFolderName('');
+    props.closeCreateFolderModal();
+  };
+
+  const CreateNewFolder = async () => {
+    if (folderName) {
+      if (/[^ A-Za-z0-9]/.test(folderName)) {
+        setCreateFolderNameValidationMessage(folderNameValidation);
+      } else {
+        setCreateFolderNameValidationMessage(undefined);
+
+        const repoId = await props.repoClient.getCurrentRepoId();
+        const postEntryChildrenRequest: PostEntryChildrenRequest =
+          new PostEntryChildrenRequest({
+            entryType: PostEntryChildrenEntryType.Folder,
+            name: folderName,
+          });
+        const requestParameters = {
+          repoId,
+          entryId: Number.parseInt(props.parentItem.id, 10),
+          request: postEntryChildrenRequest,
+        };
+        try {
+          const array = [];
+          const newFolderEntry: Entry =
+            await props.repoClient.entriesClient.createOrCopyEntry(
+              requestParameters
+            );
+
+          array.push(newFolderEntry);
+          props.closeCreateFolderModal();
+          setFolderName('');
+        } catch {
+          setCreateFolderNameValidationMessage(folderExists);

```

Comment:
There probably are but this was existing functionality. I think they copied from our code which may have been written before the API gave a useful message (if they do now) - It may be something like Object already exists